### DISCUSSION TITLE
Main menu, resolution list structure

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#10065] [Ideas]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

![image](https://user-images.githubusercontent.com/55348946/215802287-2a3f2551-4e01-4310-8fbe-eb4ad5cf34af.png)

Overall game size (especialy, when we're talking about battle and castle screens) is mostly dependent on image height. Resolutions are sorted according to the image width currently, which makes sometimes rather difficult to select the most comfortable resolution.

### Describe the possible solution

I propose to sort available resolutions according to the image height, rahter than image width.

**Comment by Laserlicht** [comment]
Or as a compromise the number of pixels (width x height). Then the aspect ratio is not the determining factor.

**Comment by Branikolog** [comment]
I personally prefer to watch resolutions sorted regarding the second number.
640x480, 1024x480 and 1920x480 preferably should be placed together, since users (in my personal opinion) after they have chosen the image height (which corresponds overall game size) want the game to fill the entire screen. I think it's logical sort list according image size (from bigger, to smaller) rather than current, since the most part of users tend to complain on UI, and battle/castle size.
Currently, If I want to play a game with maximum large battle/castle screens (on x480 resolutions), I have to search for widescreen options somewhere in the middle of the list.

**Comment by Osmodium** [comment]
I have (as just learning the codebase) looked at adding aspect ratios to the menu (using a GCD calculation based on the width and height), since I always struggle with remembering which w/h combinations equates to my screens aspect ratios 😅.
How do you feel about that as a solution?
![image](https://user-images.githubusercontent.com/234248/217068321-64cf45ae-a16a-4318-8970-3fccca06cc22.png)
EDIT: opened an improvement issue on this https://github.com/ihhub/fheroes2/issues/6586

**Comment by Laserlicht** [comment]
> How do you feel about that as a solution? 

Good idea! :)

**Comment by ihhub** [comment]
I am moving this topic into the Discussion section as this is a personal taste issue. No players expressed any concerns about the existing list of resolutions. Moreover, with changes introduced in #9323 the list of going to be even shorter.
